app/src/main/java/com/architectcoders/aacboard/ui/fragments/viewmodel/SearchPictogramsViewModel.kt

```diff
@@ -0,0 +1,70 @@
+package com.architectcoders.aacboard.ui.fragments.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.architectcoders.aacboard.data.PictogramUI
+import com.architectcoders.aacboard.data.toUIPictogram
+import com.architectcoders.aacboard.domain.data.Error
+import com.architectcoders.aacboard.domain.data.Response.Failure
+import com.architectcoders.aacboard.domain.data.Response.Success
+import com.architectcoders.aacboard.domain.use_case.location.GetLastUserRegionUseCase
+import com.architectcoders.aacboard.domain.use_case.search.SearchPictogramsUseCase
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class SearchPictogramsViewModel(
+    val searchPictogramsUseCase: SearchPictogramsUseCase,
+    val getLastUserRegionUseCase: GetLastUserRegionUseCase
+) : ViewModel() {
+
+    private val _state: MutableStateFlow<SearchPictogramUiState> = MutableStateFlow(
+        SearchPictogramUiState(),
+    )
+    val state: StateFlow<SearchPictogramUiState> get() = _state.asStateFlow()
+
+    fun onSearchClicked(searchString: String) {
+        if (searchString.isEmpty()) return
+        updateUiState(_state.value.copy(
+            searchString=searchString,
+            loading=true,
+            foundPictograms = emptyList()))
+        viewModelScope.launch {
+            val region= getLastUserRegionUseCase()
+            val response = searchPictogramsUseCase(region, searchString)
+            when (response) {
```

Comment:
Maybe we could move the region to the searchPictogramsUseCase, injecting the `RegionRepository` to the `SearchPictogramsUseCase`

Reply:
I think that in the original implementation by Fer, RegionRepository was included as one of the member variables in PictogramRepository constructor. 
I (personal opinion) decided to take it out from there since the region/language does not need to be fixed based on location (most probably it will be a user selection or a copy of the device's default language).

Background: we decided to include RegionRepository to add location/permission functionality to the project.

Open for discussion.